enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/EmbeddedInteraction.java

```diff
@@ -52,8 +52,13 @@
 
     EmbeddedInteraction( Map<String, String> config ) throws Throwable
     {
+        this (config, EphemeralFileSystemAbstraction::new);
+    }
+
+    EmbeddedInteraction( Map<String, String> config, Supplier<FileSystemAbstraction> fileSystemSupplier ) throws Throwable
+    {
         TestEnterpriseGraphDatabaseFactory factory = new TestEnterpriseGraphDatabaseFactory();
-        factory.setFileSystem( new EphemeralFileSystemAbstraction() );
+        factory.setFileSystem( fileSystemSupplier.get() );
```

Comment:
Responsibility for FSA becomes a bit unclear here. We know that the factory will inject the FSA into `PlatformModule` using the provided Supplier and that it will be added to some `life`, which means we don't need to take responsibility for closing. Although we work around this in at some place by wrapping FSA in `UncloseableDelegatingFileSystemAbstraction`.

I see that it's hard to work around this problem because some times we want to be able to keep the FSA around after interaction with a running db and some times we don't. Maybe we can document it in `TestGraphDatabaseFactory` that a provided FSA will be closed by `PlatformModule` if not wrapped in `Uncloseable...`.